### PR TITLE
Add a sentinel error for missing KV secrets

### PR DESCRIPTION
**Pull request by averche**
Adding a sentinel error value `ErrSecretNotFound`. With this change, callers can check specifically for non-existing secrets:

```go
secret, err := client.KvV2("secret").Get("foo")

if errors.Is(err, vault.ErrSecretNotFound) {
    // handle missing secret
}
if err != nil {
    // handle other errors
}
```

An example where this would be useful is when implementing an "upsert" behavior, as suggested in #16483. The caller can check the error returned from `KVv2.Patch` for `errors.Is(err, vault.ErrSecretNotFound)` and call `KvV2.Put` accordingly.

### CODE REVIEW

File: api/kv_v2.go [hunk 1]
```diff
@@ -687,18 +689,30 @@ func mergePatch(ctx context.Context, client *Client, mountPath string, secretPat
...
-		if re, ok := err.(*ResponseError); ok && re.StatusCode == 405 {
-			return readThenWrite(ctx, client, mountPath, secretPath, newData)
-		}
-
-		if re, ok := err.(*ResponseError); ok && re.StatusCode == 403 {
-			return nil, fmt.Errorf("received 403 from Vault server; please ensure that token's policy has \"patch\" capability: %w", err)
+		var re *ResponseError
+		if errors.As(err, &re) {
+			switch re.StatusCode {
+			// 403
```
Comment (digivava) [inline api/kv_v2.go:695]: This part is definitely more readable, thanks!

File: changelog/16699.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
```
Comment (digivava) [inline changelog/16699.txt:1]: Is this considered a bugfix? Or just an improvement?

**Review by digivava — APPROVED** [review]
Makes sense to me, and good improvements to some tests as well. Thank you!

File: changelog/16699.txt (hunk 2 above)
Comment (averche) [reply changelog/16699.txt:1]: Good catch, it's more of an improvement now.

**Pull request merged**